inc/Engine/Common/PerformanceHints/WarmUp/Controller.php

```diff
@@ -66,6 +66,14 @@ public function __construct( array $factories, Options_Data $options, APIClient
 	 * @return void
 	 */
 	public function warm_up_home(): void {
+		if ( 'local' === wp_get_environment_type() ) {
+			return;
+		}
+
+		if ( $this->user->is_license_expired_grace_period() ) {
+			return;
+		}
```

Comment:
Instead of duplicating those lines here and below in warm_up method, we can create a private method that has them, what do u think? we may also add the RUCSS check with them.

Reply:
Spot on 👍🏾